sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Services/WebPubSubRequestExtensions.cs

```diff
@@ -298,5 +301,25 @@ private static WebPubSubEventType GetEventType(this string ceType)
                 Constants.ContentTypes.JsonContentType => WebPubSubDataType.Json,
                 _ => throw new ArgumentException($"Invalid content type: {mediaType}")
             };
+
+        private static JsonSerializerOptions CreateSystemTextJsonSerializer()
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new ConnectionStatesConverter());
+            return options;
+        }
+
+        // support JToken for backward compatiblity.
+        private static BinaryData FromObjectAsJsonExtended<T>(T item, JsonSerializerOptions? options = null)
```

Comment:
Yes, since we opened the `SetState(string key, object value)` in old version and user probably set `JObject` which will fail when we try to convert object->BinaryData by call `BinaryData.FromObjectAsJson(T jsonSerializable)`. This is extended to cover a corner case as Newtonsoft is widely used and essential JsonSerializer in Functions.

Reply:
Right, but you're not using the type `T` anywhere.  You could just replace `T` with `object` and the code will still work fine with `JToken` or `JObject`.  It's not critical either way though.